libavcodec/qsvenc.c

```diff
@@ -823,6 +838,47 @@ static int init_video_param(AVCodecContext *avctx, QSVEncContext *q)
         q->extparam_internal[q->nb_extparam_internal++] = (mfxExtBuffer *)&q->extvsi;
     }
 
+    if (q->dual_gfx) {
+#if QSV_HAVE_HE
+        if(q->param.mfx.CodecId != MFX_CODEC_AVC && q->param.mfx.CodecId != MFX_CODEC_HEVC) {
+            av_log(avctx, AV_LOG_WARNING, "Not supported encoder for dual GFX mode. "
+                                          "Supported: h264_qsv and hevc_qsv \n");
+        }
+        if(q->param.mfx.RateControlMethod != MFX_RATECONTROL_VBR &&
+           q->param.mfx.RateControlMethod != MFX_RATECONTROL_CQP &&
+           q->param.mfx.RateControlMethod != MFX_RATECONTROL_ICQ) {
+            av_log(avctx, AV_LOG_WARNING, "Not supported BRC for dual GFX mode. "
+                                          "Supported: VBR, CQP and ICQ \n");
+        }
+        if(q->param.mfx.LowPower != MFX_CODINGOPTION_ON) {
+            av_log(avctx, AV_LOG_WARNING, "Dual GFX mode supports only low-power encoding mode \n");
+        }
+        if((q->param.mfx.CodecId == MFX_CODEC_AVC  && q->param.mfx.IdrInterval != 0) ||
```

Comment:
Commit message: "Also requires closed GOP (-idr_interval 0, it's used by default)"

Reply:
Yes, that's right. For ffmpeg user there is no difference, since ffmpeg set +1 for HEVC internally. But in this place, real values ​​are checked and for different codecs they mean different things. From the specification:

>For H.264, `IdrInterval` specifies IDR-frame interval in terms of I-frames; if `IdrInterval = 0`, then every I-frame is an IDR-frame. If `IdrInterval = 1`, then every other I-frame is an IDR-frame, etc.

>For HEVC, if `IdrInterval = 0`, then only first I-frame is an IDR-frame. If `IdrInterval = 1`, then every I-frame is an IDR-frame. If `IdrInterval = 2`, then every other I-frame is an IDR-frame, etc.